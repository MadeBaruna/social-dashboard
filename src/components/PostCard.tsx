import React from 'react';
import { Card, Button } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

interface IPostCardProps {
  id: number;
  title: string;
  body: string;
}

const PostCard = ({ id, title, body }: IPostCardProps) => (
  <Card fluid>
    <Card.Content>
      <Card.Header>{title}</Card.Header>
      <Card.Description>
        {body.split('\n').map((text, key) => (
          <span key={key}>
            {text}
            <br />
          </span>
        ))}
      </Card.Description>
    </Card.Content>
    <Card.Content extra textAlign="right">
      <Link to={`/post/${id}`}>
        <Button>Show Comments</Button>
      </Link>
      <Button icon="pencil" />
      <Button icon="trash" />
    </Card.Content>
  </Card>
);

export default PostCard;
