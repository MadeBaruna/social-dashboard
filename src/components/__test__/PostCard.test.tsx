import React from 'react';
import { shallow } from 'enzyme';

import PostCard from '../PostCard';

it('renders <PostCard /> components', () => {
  const wrapper = shallow(
    <PostCard
      id={1}
      title="this is a test title"
      body={'qui consequuntur ducimus\npossimus quisquam amet similique\nsuscipit porro ipsam amet'}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
