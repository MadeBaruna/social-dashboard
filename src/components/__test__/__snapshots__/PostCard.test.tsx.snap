// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <PostCard /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostCard
    body="qui consequuntur ducimus
possimus quisquam amet similique
suscipit porro ipsam amet"
    id={1}
    title="this is a test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CardContent>
          <CardHeader>
            this is a test title
          </CardHeader>
          <CardDescription>
            <span>
              qui consequuntur ducimus
              <br />
            </span>
            <span>
              possimus quisquam amet similique
              <br />
            </span>
            <span>
              suscipit porro ipsam amet
              <br />
            </span>
          </CardDescription>
        </CardContent>,
        <CardContent
          extra={true}
          textAlign="right"
        >
          <Link
            replace={false}
            to="/post/1"
          >
            <Button
              as="button"
            >
              Show Comments
            </Button>
          </Link>
          <Button
            as="button"
            icon="pencil"
          />
          <Button
            as="button"
            icon="trash"
          />
        </CardContent>,
      ],
      "fluid": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <CardHeader>
              this is a test title
            </CardHeader>,
            <CardDescription>
              <span>
                qui consequuntur ducimus
                <br />
              </span>
              <span>
                possimus quisquam amet similique
                <br />
              </span>
              <span>
                suscipit porro ipsam amet
                <br />
              </span>
            </CardDescription>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "this is a test title",
            },
            "ref": null,
            "rendered": "this is a test title",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span>
                  qui consequuntur ducimus
                  <br />
                </span>,
                <span>
                  possimus quisquam amet similique
                  <br />
                </span>,
                <span>
                  suscipit porro ipsam amet
                  <br />
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "qui consequuntur ducimus",
                    <br />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "qui consequuntur ducimus",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "possimus quisquam amet similique",
                    <br />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "possimus quisquam amet similique",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "suscipit porro ipsam amet",
                    <br />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "suscipit porro ipsam amet",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                ],
                "type": "span",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/post/1"
            >
              <Button
                as="button"
              >
                Show Comments
              </Button>
            </Link>,
            <Button
              as="button"
              icon="pencil"
            />,
            <Button
              as="button"
              icon="trash"
            />,
          ],
          "extra": true,
          "textAlign": "right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                as="button"
              >
                Show Comments
              </Button>,
              "replace": false,
              "to": "/post/1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": "Show Comments",
              },
              "ref": null,
              "rendered": "Show Comments",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "icon": "pencil",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "icon": "trash",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CardContent>
            <CardHeader>
              this is a test title
            </CardHeader>
            <CardDescription>
              <span>
                qui consequuntur ducimus
                <br />
              </span>
              <span>
                possimus quisquam amet similique
                <br />
              </span>
              <span>
                suscipit porro ipsam amet
                <br />
              </span>
            </CardDescription>
          </CardContent>,
          <CardContent
            extra={true}
            textAlign="right"
          >
            <Link
              replace={false}
              to="/post/1"
            >
              <Button
                as="button"
              >
                Show Comments
              </Button>
            </Link>
            <Button
              as="button"
              icon="pencil"
            />
            <Button
              as="button"
              icon="trash"
            />
          </CardContent>,
        ],
        "fluid": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <CardHeader>
                this is a test title
              </CardHeader>,
              <CardDescription>
                <span>
                  qui consequuntur ducimus
                  <br />
                </span>
                <span>
                  possimus quisquam amet similique
                  <br />
                </span>
                <span>
                  suscipit porro ipsam amet
                  <br />
                </span>
              </CardDescription>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "this is a test title",
              },
              "ref": null,
              "rendered": "this is a test title",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <span>
                    qui consequuntur ducimus
                    <br />
                  </span>,
                  <span>
                    possimus quisquam amet similique
                    <br />
                  </span>,
                  <span>
                    suscipit porro ipsam amet
                    <br />
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "qui consequuntur ducimus",
                      <br />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "qui consequuntur ducimus",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "possimus quisquam amet similique",
                      <br />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "possimus quisquam amet similique",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "suscipit porro ipsam amet",
                      <br />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "suscipit porro ipsam amet",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                  ],
                  "type": "span",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/post/1"
              >
                <Button
                  as="button"
                >
                  Show Comments
                </Button>
              </Link>,
              <Button
                as="button"
                icon="pencil"
              />,
              <Button
                as="button"
                icon="trash"
              />,
            ],
            "extra": true,
            "textAlign": "right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  as="button"
                >
                  Show Comments
                </Button>,
                "replace": false,
                "to": "/post/1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "button",
                  "children": "Show Comments",
                },
                "ref": null,
                "rendered": "Show Comments",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "icon": "pencil",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "icon": "trash",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
